src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,25 @@
+import { StringParam, useQueryParams } from 'use-query-params';
+
+type Props = {
+  sortBy: string,
+};
+
+export const Icon: React.FC<Props> = ({ sortBy }) => {
+  const [searchParams] = useQueryParams({
+    sort: StringParam,
+    order: StringParam,
+  });
+
+  const isSorting = searchParams.sort === sortBy;
+  const isDesc = searchParams.order === 'desc';
```

Comment:
It's better to create an enum for sort orders to avoid errors connected with typos